### DISCUSSION TITLE
Dynamically exposing schemas in Supabase

### DISCUSSION THREAD

**Discussion by dinodeSimon** [supabase/supabase#30661] [Questions]
Hi,
Is it possible to have all schemas matching a pattern exposed in Supabase? I've tried using the config described in the postgerest documentation but it doesn't appear to work. 

Basically I want all api_* schemas exposed and as new ones are added a trigger to reload the config.

Not sure if i'm doing something wrong or it isn't supported
https://postgrest.org/en/stable/references/api/schemas.html

**Comment by dinodeSimon** [comment]
There does appear to be a supported way to do it via the management api but ideally i'd like to avoid having to do that but would still like to see the results reflected in the dashboard
https://supabase.com/docs/reference/api/v1-update-postgrest-service-config

**Comment by dinodeSimon** [comment]
Just another follow-up. This does appear to work although it is not reflected in the dashboard which is not ideal

`DO $$
DECLARE
    schema_list TEXT;
BEGIN
    -- Generate the list of schemas
    SELECT string_agg(nspname, ',')
    INTO schema_list
    FROM pg_namespace
    WHERE nspname LIKE 'api_%';

    EXECUTE format('ALTER ROLE authenticator SET pgrst.db_schemas = %L', schema_list);
END $$;

NOTIFY pgrst, 'reload config';
NOTIFY pgrst, 'reload schema';
`

**Comment by Garett-MacGowan** [comment]
I used to be able to do this with a postgres function exposed through the auto generated API, but `ALTER ROLE authenticator SET pgrst.db_schemas` no longer works with the service_role. I'm thinking of giving https://supabase.com/docs/reference/api/v1-update-postgrest-service-config a try. Do you know if changes made through this endpoint reflect in the UI? I assume they do.

**Comment by Garett-MacGowan** [comment]
@dinodeSimon I verified that the `v1-update-postgrest-service-config` endpoint updates the entries in the dashboard.
